llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4552,7 +4555,7 @@ LoopVectorizationPlanner::selectInterleaveCount(VPlan &Plan, ElementCount VF,
   // We don't attempt to perform interleaving for loops with uncountable early
   // exits because the VPInstruction::AnyOf code cannot currently handle
   // multiple parts.
-  if (Plan.hasEarlyExit())
+  if (Plan.hasEarlyExit() || Plan.shouldEarlyExitContinueInScalarLoop())
```

Comment:
From what I can see in the tests the plan does in fact have an early exit, so we shouldn't need to ask the question `shouldEarlyExitContinueInScalarLoop`. It's very confusing for a plan that has an early exit to return false from `hasEarlyExit`